rusqlite_migration/src/tests/synch.rs

```diff
@@ -0,0 +1,472 @@
+}
+
+#[test]
+fn test_migration_definition_error_display() {
+    let err = MigrationDefinitionError::DownNotDefined { migration_index: 1 };
+    assert_eq!(
+        "Migration 1 (version 1 -> 2) cannot be reverted",
+        format!("{err}")
+    );
+
+    let err = MigrationDefinitionError::DatabaseTooFarAhead;
+    assert_eq!(
+        "Attempt to migrate a database with a migration number that is too high",
+        format!("{err}")
+    );
+
+    let err = MigrationDefinitionError::NoMigrationsDefined;
+    assert_eq!(
+        "Attempt to migrate with no migrations defined",
+        format!("{err}")
+    )
+}
+
+#[test]
+fn test_error_display() {
+    let err = Error::SpecifiedSchemaVersion(SchemaVersionError::TargetVersionOutOfRange {
+        specified: SchemaVersion::NoneSet,
+        highest: SchemaVersion::NoneSet,
+    });
+    assert_eq!(
+        "rusqlite_migrate error: SpecifiedSchemaVersion(TargetVersionOutOfRange { specified: NoneSet, highest: NoneSet })",
+        format!("{err}")
+    );
+
+    let err = Error::Hook(String::new());
+    assert_eq!("rusqlite_migrate error: Hook(\"\")", format!("{err}"));
+
+    let err = Error::ForeignKeyCheck(ForeignKeyCheckError {
+        table: String::new(),
+        rowid: 1,
+        parent: String::new(),
+        fkid: 2,
+    });
+    assert_eq!("rusqlite_migrate error: ForeignKeyCheck(ForeignKeyCheckError { table: \"\", rowid: 1, parent: \"\", fkid: 2 })", format!("{err}"));
+
+    let err = Error::MigrationDefinition(MigrationDefinitionError::NoMigrationsDefined);
+    assert_eq!(
+        "rusqlite_migrate error: MigrationDefinition(NoMigrationsDefined)",
+        format!("{err}")
+    );
+
+    let err = Error::RusqliteError {
+        query: String::new(),
+        err: rusqlite::Error::InvalidQuery,
+    };
+    assert_eq!(
+        "rusqlite_migrate error: RusqliteError { query: \"\", err: InvalidQuery }",
+        format!("{err}")
+    );
+}
+
+#[test]
+fn schema_version_partial_display_test() {
+    assert_eq!("0 (no version set)", format!("{}", SchemaVersion::NoneSet));
+    assert_eq!(
+        "1 (inside)",
+        format!("{}", SchemaVersion::Inside(NonZeroUsize::new(1).unwrap()))
+    );
+    assert_eq!(
+        "32 (inside)",
+        format!("{}", SchemaVersion::Inside(NonZeroUsize::new(32).unwrap()))
+    );
+    assert_eq!(
+        "1 (outside)",
+        format!("{}", SchemaVersion::Outside(NonZeroUsize::new(1).unwrap()))
+    );
+    assert_eq!(
+        "32 (outside)",
+        format!("{}", SchemaVersion::Outside(NonZeroUsize::new(32).unwrap()))
+    );
+}
+
+#[test]
+fn error_test_source() {
+    let err = Error::RusqliteError {
+        query: String::new(),
+        err: rusqlite::Error::InvalidQuery,
+    };
+    assert!(
+        match std::error::Error::source(&err).and_then(|e| e.downcast_ref::<rusqlite::Error>()) {
+            Some(e) => e == &rusqlite::Error::InvalidQuery,
+            None => false,
+        }
+    );
```

Comment:
I’m not the biggest fan of the `assert!(match …)` construct when something simpler is possible. I realize it’s a matter of taste and that some instance may exist in the rest of the code, but it’s probably a good chance to replace them with something like:

```suggestion
    assert!(std::error::Error::source(&err).and_then(|e| e.downcast_ref::<rusqlite::Error>()).unwrap(),  &rusqlite::Error::InvalidQuery);
```
(would need to be tested and `rustfmt`’d)

What do you think?

Reply:
Thanks! I was unsure how to simplify this :D.